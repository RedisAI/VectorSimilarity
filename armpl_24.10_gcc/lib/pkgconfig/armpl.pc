armpl-static-lp64-seq.pc