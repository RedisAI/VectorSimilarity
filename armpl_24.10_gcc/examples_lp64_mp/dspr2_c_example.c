/*
 * Double precision symmetric rank 2 operation in packed format
 *
 * Arm Performance Libraries version 24.10
 * SPDX-FileCopyrightText: Copyright 2015-2024 Arm Limited and/or its affiliates
 */

#include <stdio.h>
#include <stdlib.h>
#include "armpl.h"

#define N 4

int main()
{
	const char uplo = 'U';

	const int n = N;
	const int incx = 1;
	const int incy = 1;

	const double alpha = 1.1;

	double ap[N*(N+1)/2], x[N], y[N];

	printf("ARMPL example: double precision symmetric rank 2 operation using DSPR2\n");
	printf("----------------------------------------------------------------------\n");
	printf("\n");

	/* Initialize matrix A in packed format */
	for (int i = 0; i < n*(n+1)/2; ++i) {
		ap[i] = (double)(i+1);
	}

	/* Initialize vector x and y */
	for (int i = 0; i < n; ++i) {
		x[i] = (double)(2*i+1);
		y[i] = (double)(3*i+1);
	}

	printf("alpha : %5.3F\n", alpha);
	printf("\n");

	printf("Matrix A:\n");
	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < n; ++j) {
			int idx = j < i ? j + i*(i+1)/2 : i + j*(j+1)/2;
			printf(" %13.3E", ap[idx]);
		}
		printf("\n");
	}

	printf("\n");
	printf("Vector x:\n");
	for (int i = 0; i < n; ++i) {
		printf(" %13.3E\n", x[i]);
	}

	printf("\n");
	printf("Vector y:\n");
	for (int i = 0; i < n; ++i) {
		printf(" %13.3E\n", y[i]);
	}

	dspr2_(&uplo, &n, &alpha, &x[0], &incx, &y[0], &incy, &ap[0]);

	printf("\n");
	printf("Updated matrix A = alpha*x*y**T + alpha*y*x**T + A:\n");
	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < n; ++j) {
			int idx = j < i ? j + i*(i+1)/2 : i + j*(j+1)/2;
			printf(" %13.3E", ap[idx]);
		}
		printf("\n");
	}

	return 0;
}
