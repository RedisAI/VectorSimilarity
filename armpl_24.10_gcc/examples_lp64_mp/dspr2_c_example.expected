ARMPL example: double precision symmetric rank 2 operation using DSPR2
----------------------------------------------------------------------

alpha : 1.100

Matrix A:
     1.000E+00     2.000E+00     4.000E+00     7.000E+00
     2.000E+00     3.000E+00     5.000E+00     8.000E+00
     4.000E+00     5.000E+00     6.000E+00     9.000E+00
     7.000E+00     8.000E+00     9.000E+00     1.000E+01

Vector x:
     1.000E+00
     3.000E+00
     5.000E+00
     7.000E+00

Vector y:
     1.000E+00
     4.000E+00
     7.000E+00
     1.000E+01

Updated matrix A = alpha*x*y**T + alpha*y*x**T + A:
     3.200E+00     9.700E+00     1.720E+01     2.570E+01
     9.700E+00     2.940E+01     5.010E+01     7.180E+01
     1.720E+01     5.010E+01     8.300E+01     1.179E+02
     2.570E+01     7.180E+01     1.179E+02     1.640E+02
